### PR TITLE
3.2.0 API changes page 

### PR DESCRIPTION
**Pull request by ImportanceOfBeingErnest**
## PR Summary

New version of the 3.2.0 API changes page. This was started by @dstansby in #15141. Here, the remaining API changes are incorporated. 

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by ImportanceOfBeingErnest** [comment]
I'm not 100% confident this is complete. E.g. in #11387 a file `doc/api/next_api_changes/2018-10-04-AL-deprecations.rst` came in, but it's not in current master.

**Comment by tacaswell** [comment]
Good catch on that!  I am almost afraid to ask how you noticed that, see https://github.com/matplotlib/matplotlib/pull/11387#issuecomment-526929527 for what I think is going on here.  

I think we should bump the warning to v3.2 in the code and duplicate the deprecation text in the 3.2 API changes, but not worry about removing it from the 3.1 docs

**Comment by ImportanceOfBeingErnest** [comment]
I now did duplicate the deprecation text in the 3.2 API changes. I did not touch source code here.

There is another "conflict" for api/api_changes/2019-02-11-PGE.rst from https://github.com/matplotlib/matplotlib/pull/13986/files. That's a change to an entry that initially into 3.1, but the change did not get backported. I think this change is anyways rather a "what's new" (now more things are supported), so we could mention it in the what's new of 3.2 instead.

**Comment by dstansby** [comment]
Thanks for picking this up!

File: doc/api/prev_api_changes/api_changes_3.2.0/behavior.rst [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+``sticky_edges`` now have an additional effect on margins application: even if
+an axis limit did not coincide with a sticky edge, it cannot *cross* a sticky
+edge through margin application -- instead, the margins will only expand the
+axis limit until it bumps against the sticky edge.
+
+This change improves the margins of axes displaying a `~Axes.streamplot`:
+
+- if the streamplot goes all the way to the edges of the vector field, then the
+  axis limits are set to match exactly the vector field limits (whereas they
+  would be sometimes be off by a small floating point error previously).
```
[the remark this answers is not in the document]
Comment (NelleV) [reply doc/api/prev_api_changes/api_changes_3.2.0/behavior.rst]: Fixed.

File: doc/api/prev_api_changes/api_changes_3.2.0/behavior.rst [hunk 2]
```diff
@@ -0,0 +1,263 @@
+
+Behavior changes
+----------------
+
+Reduced default value of :rc:`axes.formatter.limits`
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
```
[the remark this answers is not in the document]
Comment (NelleV) [reply doc/api/prev_api_changes/api_changes_3.2.0/behavior.rst]: I didn't fix this, as we do not have a rule about that and it is done inconsistently in our documentation. If we are to establish a rule, it should be done in such a manner that everyone is aware.

File: doc/api/prev_api_changes/api_changes_3.2.0/behavior.rst [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+axis limit until it bumps against the sticky edge.
+
+This change improves the margins of axes displaying a `~Axes.streamplot`:
+
+- if the streamplot goes all the way to the edges of the vector field, then the
+  axis limits are set to match exactly the vector field limits (whereas they
+  would be sometimes be off by a small floating point error previously).
+- if the streamplot does not reach the edges of the vector field (e.g., due to
+  the use of ``start_points`` and ``maxlength``), then margins expansion will
+  not cross the the vector field limits anymore.
```
[the remark this answers is not in the document]
Comment (NelleV) [reply doc/api/prev_api_changes/api_changes_3.2.0/behavior.rst]: Fixed

File: doc/api/prev_api_changes/api_changes_3.2.0/behavior.rst [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+
+This change is also used internally to ensure that polar plots don't display
```
[the remark this answers is not in the document]
Comment (NelleV) [reply doc/api/prev_api_changes/api_changes_3.2.0/behavior.rst]: Didn't fix this. I don't think it matters much.

File: doc/api/prev_api_changes/api_changes_3.2.0/behavior.rst [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+resampling the data in A for display (or saving to file) *if* the upsample
+rate is less than a factor of three, and not an integer; downsampled data is
+always smoothed at resampling.
+
+To get the old behavior, set :rc:`interpolation` to the old default "nearest"
+(or specify the ``interpolation`` kwarg of `.Axes.imshow`)
+
+To always get the anti-aliasing behavior, no matter what the up/down sample
+rate, set :rc:`interpolation` to "hanning" (or one of the other filters
+available.
```
[the remark this answers is not in the document]
Comment (NelleV) [reply doc/api/prev_api_changes/api_changes_3.2.0/behavior.rst]: Fixed.

File: doc/api/prev_api_changes/api_changes_3.2.0/deprecations.rst [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+or ``Axis.minor.formatter`` to an object that is not a subclass of `Locator` or
+`Formatter` (respectively) is deprecated.  Note that these attributes should
+usually be set using `Axis.set_major_locator`, `Axis.set_minor_locator`, etc.
+which already raise an exception when an object of the wrong class is passed.
+
+Passing more than one positional argument or unsupported keyword arguments to
+`~matplotlib.axes.Axes.axis()` is deprecated (such arguments used to be
+silently ignored).
+
+``minor`` argument will be become keyword-only
```
[the remark this answers is not in the document]
Comment (NelleV) [reply doc/api/prev_api_changes/api_changes_3.2.0/deprecations.rst]: Fixed.

[the remark this answers is not in the document]
Comment (NelleV) [reply doc/api/prev_api_changes/api_changes_3.2.0/deprecations.rst]: (That's definitely not proper english)

**Pull request merged**